server/sonar-web/src/main/js/app/components/AdminContainer.tsx

```diff
@@ -17,35 +17,52 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-import React from 'react';
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
 import Helmet from 'react-helmet';
 import { connect } from 'react-redux';
 import SettingsNav from './nav/settings/SettingsNav';
 import { getAppState } from '../../store/rootReducer';
-import { onFail } from '../../store/rootActions';
 import { getSettingsNavigation } from '../../api/nav';
 import { setAdminPages } from '../../store/appState/duck';
 import { translate } from '../../helpers/l10n';
+import { Extension } from '../types';
+
+interface Props {
+  appState: {
+    adminPages: Extension[];
+    organizationsEnabled: boolean;
+  };
+  location: {};
+  setAdminPages: (adminPages: Extension[]) => void;
+}
+
+class AdminContainer extends React.PureComponent<Props> {
+  static contextTypes = {
+    canAdmin: PropTypes.bool.isRequired
+  };
 
-class AdminContainer extends React.PureComponent {
   componentDidMount() {
-    if (!this.props.appState.canAdmin) {
+    if (!this.context.canAdmin) {
       // workaround cyclic dependencies
       const handleRequiredAuthorization = require('../utils/handleRequiredAuthorization').default;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') require() style import is forbidden [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3533)